pkg/build/buildkit/deploy.go

```diff
@@ -27,14 +27,35 @@ func Exists(ctx context.Context, c client.Client) (bool, error) {
 	} else if err != nil {
 		return false, err
 	}
+
+	// check if the controller is running outside of a deployment
+	err = checkControllerDeployment(ctx, c)
```

Comment:
I still think checking for the env var being set is a cleaner approach and abstracts you away from having to know about "how" the controller was deployed, but i want to get this in, so merging as is.